packages/storefront-lwc/src/modules/commerce/basket/basket.js

```diff
@@ -4,30 +4,36 @@
     SPDX-License-Identifier: BSD-3-Clause
     For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */
-import { LightningElement } from 'lwc';
-import { ShoppingBasket } from 'commerce/data';
+import { LightningElement, wire } from 'lwc';
+import { GET_BASKET, REMOVE_ITEM_FROM_BASKET } from './gql.js';
+import { useMutation, useQuery } from '@lwce/apollo-client';
 
 export default class Basket extends LightningElement {
     products = [];
     loading = true;
     basket = {};
+    shippingMethods = [];
+    selectedShippingMethodId;
 
-    get hasProducts() {
-        return this.products.length > 0;
-    }
-
-    constructor() {
-        super();
-        ShoppingBasket.updateBasketListener(
-            this.updateBasketHandler.bind(this),
-        );
+    @wire(useQuery, {
+        query: GET_BASKET,
+        lazy: false,
+    })
+    getBasket(response) {
+        this.loading = response.loading;
+        if (response.initialized) {
+            this.basket = response.data.getBasket || {};
+            this.shippingMethods = this.basket.shippingMethods.applicableShippingMethods;
+            this.selectedShippingMethodId = this.basket.selectedShippingMethodId;
+            this.shippingMethods = this.filterStorePickupShippingMethods(
+                this.basket.shippingMethods.applicableShippingMethods,
+            );
+            this.products = [...this.basket.products];
```

Comment:
NIT picky, but is the destructuring necessary here? `this.basket.products` will already be a new array from line 25

Reply:
Good point. Missed that. Will fix